src/sql.rs

```diff
@@ -299,28 +303,47 @@ impl Sql {
         }
     }
 
+    /// Locks the write transactions mutex. We use DEFERRED transactions for more parallelism, but
+    /// then db writers should be serialised, otherwise a write transaction A can retry for an
+    /// indefinite time and then fail by timeout if other writers modified the db since A
+    /// started. So, we sacrifice some parallelism for doing less work and more fairness in
+    /// servicing transactions.
+    pub async fn write_lock(&self) -> MutexGuard<'_, ()> {
+        self.wlk.lock().await
+    }
+
     /// Allocates a connection and calls given function with the connection.
     ///
     /// Returns the result of the function.
-    pub async fn call<'a, F, R>(&'a self, function: F) -> Result<R>
+    pub async fn call<F, R>(&self, function: F) -> Result<R>
```

Comment:
If calling this with a write query can cause crashes, would be nice if it was called something that makes it more clear that it should only be used for writes - sth. like `read` (and then rename `callw` -> `write` or `readwrite`), `call_readonly`, or `call_ro` (and `call_rw`).

Reply:
If you call this for a write query, it may fail if another write query is done in parallel. Or that another write query may fail of course. I can rename callw() to call_rw() if you find more expressive. But as call_rw() do call() inside, i'd avaoid renaming it to call_ro() because it would look strange.

Reply:
> But as call_rw() do call() inside, i'd avaoid renaming it to call_ro() because it would look strange.

You could create a third, non-public, function which is called `call()`, and then `call_rw` an `call_ro` both call this third function internally.

SQL is used all over the place, and quite sometimes we have new contributors who won't know that you must not use `call()` if you want to write, leading to SQLITE_BUSY failures.

Reply:
Yet another idea: `call` isn't used outside of the sql module at all, so you could make it private, and simply mention in the doc comment that it must not be used for writing and why and that `callw` should be used for that. Then the footgun is not gone completely, but restrained to one file.